src/lib/grad.jl

```diff
@@ -4,7 +4,7 @@ macro nograd(ex)
   isexpr(ex, :tuple) || (ex = Expr(:tuple, ex))
   blk = @q begin end
   for f in ex.args
-    back = MacroTools.@q _ -> ($__source__; nothing)
+    back = MacroTools.@q _ -> ($__source__; Zero())
```

Comment:
Probably better is `DoesNotExist`
`@nograd` is mostly used for things where the gradient makes no sense, not that is zero valued.
```suggestion
    back = MacroTools.@q _ -> ($__source__; DoesNotExist())
```